pkg/generator/ingress_translator_test.go

```diff
@@ -891,6 +891,73 @@ func TestIngressTranslatorWithUpstreamTLS(t *testing.T) {
 				internalVirtualHosts:    vHosts,
 			}
 		}(),
+	}, {
+		name: "http2 and https",
```

Comment:
The test passes even if we do:

```
for _, port := range service.Spec.Ports {
  if port.Port == split.ServicePort.IntVal || port.Name == split.ServicePort.StrVal {
    externalPort = port.Port
    targetPort = port.TargetPort.IntVal
  }
  if port.Name == "http2" || port.Name == "h2c" {
    http2 = true
  }
}
```
Assignment `http2 = port.Name == "http2" || port.Name == "h2c"` was problematic because it would override the previous value.
Is there a scenario where:
`if port.Port != split.ServicePort.IntVal && port.Name != split.ServicePort.StrVal`
and `if port.Name == "http2" || port.Name == "h2c"`?

Reply:
The logic is one of service port has `http2` port, we use `http2`. So it is not problematic.

Reply:
> The logic is that one of service port has http2 port, then we use http2. So it is not problematic.
Yes I understand that, but since the test passes we should keep the logic stricter and not move that outside the loop.
My question still holds what is the gain of moving that part outside the first condition?

Reply:
Okay, added one more test case for "http and https".

Reply:
@skonto If you have some suggestion, could you please provide the code snippet?

Reply:
> When the service has two ports (e.g. http2 and https) but Ingress has one port 443, the scenario you mentioned happens.

Do you have a test case for that? If the test case is `name: "http2 and https"`, then as I mentioned above having the second within the first still makes the test pass:
```
for _, port := range service.Spec.Ports {
  if port.Port == split.ServicePort.IntVal || port.Name == split.ServicePort.StrVal {
    externalPort = port.Port
    targetPort = port.TargetPort.IntVal
  }
  if port.Name == "http2" || port.Name == "h2c" {
    http2 = true
  }
}
```
So I dont think we test then the case where `port.Port != split.ServicePort.IntVal && port.Name != split.ServicePort.StrVal` is true. Reason is port.Port == split.ServicePort.IntVal is always true in this test. Both equal to zero.

Reply:
A scenario where you actually test this is:
```
		state: []runtime.Object{
			svc("servicens", "servicename", func(service *corev1.Service) {
				service.Spec.Ports = []corev1.ServicePort{{
					Name:       "http2",
					TargetPort: intstr.FromInt(8080),
					**Port: 80,**
				}, {
					Name:       "https",
					TargetPort: intstr.FromInt(443),
				}}
			}),
			eps("servicens", "servicename"),
			caSecret,
		},
```

Reply:
Okay, so you wanted to suggest the missing port in the test case. Thank you. Updated.
And the snippet in the initial comment https://github.com/knative-sandbox/net-kourier/pull/813#discussion_r840440109, I guess it is a mistake and you probably wanted to wirte the snippet in https://github.com/knative-sandbox/net-kourier/pull/813#discussion_r840470277 :thinking:

Reply:
Yes :) Sorry it was a bit hard to explain this. :+1:

Reply:
Nah, it's my bad. Thank you for the review :bow: